proxy-node-gateway/src/test/java/uk/gov/ida/notification/resources/HubResponseResourceTest.java

```diff
@@ -108,16 +108,40 @@ public void testsHappyPath() {
             .collect(Collectors.toList());
 
         List<String> expectedLogOutput = Lists.newArrayList(
-            "[HUB Response] received for hub authn request ID 'hub_request_id_in_session', eIDAS authn request ID 'eidas_request_id_in_session'",
-            "[eIDAS Response] received for hub authn request ID 'hub_request_id_in_session', eIDAS authn request ID 'eidas_request_id_in_session'"
+            "[HUB Response] received for session 'session-id', hub authn request ID 'hub_request_id_in_session', eIDAS authn request ID 'eidas_request_id_in_session'",
+            "[eIDAS Response] received for session 'session-id', hub authn request ID 'hub_request_id_in_session', eIDAS authn request ID 'eidas_request_id_in_session'"
         );
 
         assertEquals(expectedLogOutput, allLogRecords);
     }
 
     @Test
-    public void shouldThrowSamlAttributeErrorIfMissingSessionAttributes() {
-        when(session.getAttribute(SESSION_KEY_HUB_REQUEST_ID)).thenReturn(null);
+    public void shouldThrowSamlAttributeErrorIfMissingSessionIsNull() {
+        when(session.getAttribute(SESSION_KEY_SESSION_DATA)).thenReturn(null);
+
+        HubResponseResource resource =  new HubResponseResource(
+            new SamlFormViewBuilder(),
+            translatorProxy
+        );
+
+        try {
```

Comment:
https://github.com/alphagov/verify-proxy-node/blob/15660594557f64bcaf92862c523bdeca63d8fca8/proxy-node-gateway/src/test/java/uk/gov/ida/notification/resources/HubResponseResourceTest.java#L127-L132

When testing for exceptions thrown, can we please use the mechanism provided by the testing framework? AssertJ has a mechanism to capture and check expected properties - these should be used instead of try/catch. Examples:

```
assertThatThrownBy(() -> { throw new Exception("boom!"); }).isInstanceOf(Exception.class)
                                                             .hasMessageContaining("boom");
```
```
assertThatExceptionOfType(IOException.class).isThrownBy(() -> ...
```
```
   // when
   Throwable thrown = catchThrowable(() -> { throw new Exception("boom!"); });

   // then
   assertThat(thrown).isInstanceOf(Exception.class)
```
Guidance here: http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#exception-assertion

Can we please re-write these tests to use the testing framework methods?

Reply:
Nice - i'll get these changed

Reply:
👍